go/cmd/dolt/commands/rebase.go

```diff
@@ -0,0 +1,369 @@
+func buildInitialRebaseMsg(sqlCtx *sql.Context, queryist cli.Queryist, rebaseBranch, currentBranch string) (string, error) {
+	var buffer bytes.Buffer
+
+	rows, err := GetRowsForSql(queryist, sqlCtx, "SELECT action, commit_hash, commit_message FROM dolt_rebase ORDER BY rebase_order")
+	if err != nil {
+		return "", err
+	}
+
+	// rebase plan
+	for _, row := range rows {
+		action, found := getRebaseAction(row[0])
+		if !found {
+			return "", errors.New("invalid rebase action")
+		}
+		commitHash := row[1].(string)
+		commitMessage := row[2].(string)
+		buffer.WriteString(fmt.Sprintf("%s %s %s\n", action, commitHash, commitMessage))
+	}
+	buffer.WriteString("\n")
+
+	// help text
+	rebaseBranchHash, err := getHashOf(queryist, sqlCtx, rebaseBranch)
+	if err != nil {
+		return "", err
+	}
+	currentBranchHash, err := getHashOf(queryist, sqlCtx, currentBranch)
+	if err != nil {
+		return "", err
+	}
+	rows, err = GetRowsForSql(queryist, sqlCtx, "SELECT COUNT(*) FROM dolt_rebase")
+	if err != nil {
+		return "", err
+	}
+	numSteps, err := getInt64ColAsInt64(rows[0][0])
+	if err != nil {
+		return "", err
+	}
+	buffer.WriteString(fmt.Sprintf("# Rebase %s..%s onto %s (%d commands)\n#\n", rebaseBranchHash, currentBranchHash, rebaseBranchHash, numSteps))
+
+	buffer.WriteString("# Commands:\n")
+	buffer.WriteString("# p, pick <commit> = use commit\n")
+	buffer.WriteString("# d, drop <commit> = remove commit\n")
+	buffer.WriteString("# r, reword <commit> = use commit, but edit the commit message\n")
+	buffer.WriteString("# s, squash <commit> = use commit, but meld into previous commit\n")
+	buffer.WriteString("# f, fixup <commit> = like \"squash\", but discard this commit's message\n")
+	buffer.WriteString("# These lines can be re-ordered; they are executed from top to bottom.\n")
+	buffer.WriteString("#\n")
+	buffer.WriteString("# If you remove a line here THAT COMMIT WILL BE LOST.\n")
+	buffer.WriteString("#\n")
+	buffer.WriteString("# However, if you remove everything, the rebase will be aborted.\n")
+	buffer.WriteString("#\n")
+
+	return buffer.String(), nil
+}
+
+// getRebaseAction returns the rebase action for the given row. This conversion is necessary because a local client
+// returns an int representing the enum whereas a remote client properly returns the label.
+// TODO: Remove this once the local client returns the label.
+func getRebaseAction(col interface{}) (string, bool) {
+	action, ok := col.(string)
+	if ok {
+		return action, true
+	} else {
+		return dprocedures.RebaseActionEnumType.At(int(col.(uint16)))
+	}
+}
+
+// parseRebaseMessage parses the rebase message from the editor and adds all uncommented out lines as steps in the rebase plan.
+func parseRebaseMessage(rebaseMsg string) (*rebase.RebasePlan, error) {
+	plan := &rebase.RebasePlan{}
+	splitMsg := strings.Split(rebaseMsg, "\n")
+	for i, line := range splitMsg {
+		if !strings.HasPrefix(line, "#") && strings.ReplaceAll(line, " ", "") != "" {
```

Comment:
[strings.TrimSpace](https://pkg.go.dev/strings#TrimSpace) would probably be a better fit here. That will remove all leading and trailing whitespace (including tabs)